### PR TITLE
[4.0] Add module to dashboard - save and close button

### PR DESCRIPTION
**Pull request by rjharishabh**
Pull Request for Issue #33630 .

### Summary of Changes
https://github.com/joomla/joomla-cms/issues/33630#issuecomment-835691242

### Testing Instructions
Go to a dashboard, click "add a module".
Select a module, for example, logged-users
A Modal opens for module settings

### Actual result BEFORE applying this Pull Request
before-module

### Expected result AFTER applying this Pull Request
after-module

### Documentation Changes Required
None

### CODE REVIEW

**Comment by chmst** [comment]
Thank you for the fast work - it is good for modules. But now we have a save and close too much 

grafik

**Comment by rjharishabh** [comment]
> But now we have a save and close too much

@chmst Sorry, but I didn't  understand your comment

**Comment by richard67** [comment]
> > But now we have a save and close too much
> 
> @chmst Sorry, but I didn't understand comment

@rjharishabh See the screenshot above. The modal to add new modules has a "Save & Close" button now, but I think it shouldn't.

**Comment by rjharishabh** [comment]
> @rjharishabh See the screenshot above. The modal to add new modules has a "Save & Close" button now, but I think it shouldn't.

means  `Close` and `Save` should be there,  not `Save & Close`

**Comment by chmst** [comment]
Only close.
On this screen, we can selct something, but there is nothing to save.

**Comment by rjharishabh** [comment]
Working on a fix

**Comment by rjharishabh** [comment]
It's an important issue, So it should be resolved very quickly and I am a newbie to Joomla, will take more time than required.
Thanks

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 016ce8dca12d217d2286308249cf83cef521b3b4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33631">issues.joomla.org/tracker/joomla-cms/33631</a>.</sub>

**Comment by rjharishabh** [comment]
@joomdonation code review please

File: administrator/components/com_modules/tmpl/module/modal.php [hunk 1]
```diff
@@ -9,6 +9,17 @@
 
 defined('_JEXEC') or die;
 
+use Joomla\CMS\Factory;
+
+/** @var \Joomla\CMS\WebAsset\WebAssetManager $wa */
+$wa = Factory::getApplication()->getDocument()->getWebAssetManager();
+$wa->addInlineScript('
+    document.addEventListener("DOMContentLoaded", function() {
+        saveCloseButton = window.parent.document.getElementById("btnModalSaveAndClose");
```
Comment (joomdonation) [inline administrator/components/com_modules/tmpl/module/modal.php]: Could you please change it to:
```javascript
constant saveCloseButton = window.parent.document.getElementById("btnModalSaveAndClose");
```

And maybe to be safe, we should add an if command to check and make sure the button exist before calling remove class

Something like:
```javascript
constant saveCloseButton = window.parent.document.getElementById("btnModalSaveAndClose");
if (saveCloseButton)
{
     saveCloseButton.classList.remove("hidden");
}
```

Comment (rjharishabh) [reply administrator/components/com_modules/tmpl/module/modal.php]: Done
Thanks @joomdonation

**Comment by joomdonation** [comment]
@rjharishabh Look good now. Thanks for working on it. As soon as the codestyle check complete, I will mark my test result and restore @sandramay0905 test result so that it is RTC

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on afe077613cb546f9637e612f39b9c4a5da96a762<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33631">issues.joomla.org/tracker/joomla-cms/33631</a>.</sub>

**Comment by joomdonation** [comment]
RTC (Try to see if I can mark this as RTC :D )<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33631">issues.joomla.org/tracker/joomla-cms/33631</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33631">issues.joomla.org/tracker/joomla-cms/33631</a>.</sub>

**Comment by PhilETaylor** [comment]
Sorry to be late to the party, but close/cancel seems to always be the last (right most) button in other modals and toolbars, can we quickly switch these buttons around please for consistency?

**Comment by richard67** [comment]
Back to pending.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33631">issues.joomla.org/tracker/joomla-cms/33631</a>.</sub>

**Comment by richard67** [comment]
@rjharishabh I think @PhilETaylor is right and we should change order of these two buttons for consistency. Could you fix that?

**Comment by joomdonation** [comment]
Or we can merge this and ask him to make a new PR so that testers don't have to test again. That should work.

**Comment by PhilETaylor** [comment]
merge it!

**Comment by richard67** [comment]
> Or we can merge this and ask him to make a new PR so that testers don't have to test again. That should work.

If this PR needs 2 tests again or if another PR needs 2 tests doesn't make a difference.

For me it's important that it is not forgotten. If I merge this now, I need a new issue for the order.

**Comment by richard67** [comment]
RTC again. @PhilETaylor Please open a new issue for the ordering when this has been merged.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33631">issues.joomla.org/tracker/joomla-cms/33631</a>.</sub>

**Comment by PhilETaylor** [comment]
In trying prove that cancel/close was the last button I found other modals where that was not the case, so there are a few places we need to change (unrelated to this PR)

**Comment by richard67** [comment]
I thought we had cleaned that up recently.

**Comment by PhilETaylor** [comment]
> I thought we had cleaned that up recently.

Me too.

**Comment by richard67** [comment]
Thanks!

**Comment by rjharishabh** [comment]
Thanks

**Pull request merged**